Libraries/LibRegex/RegexParser.cpp

```diff
@@ -993,6 +993,12 @@ bool ECMA262Parser::has_duplicate_in_current_alternative(FlyString const& name)
 
 bool ECMA262Parser::parse_disjunction(ByteCode& stack, size_t& match_length_minimum, ParseFlags flags)
 {
+    // Check recursion depth to prevent stack overflow
+    if (m_recursion_depth >= MAX_RECURSION_DEPTH)
+        return set_error(Error::ReachedMaxRecursion);
+
+    ++m_recursion_depth;
```

Comment:
This + the `--`s somewhere down there:
```suggestion
    TemporaryChange recursion_depth_change { m_recursion_depth, m_recursion_depth + 1 };
```